### PR TITLE
feat(FOROME-790): add new UI for step selecting

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-790

The logic for editing attributes from the right panel is coming soon!

### CODE REVIEW

File: src/pages/filter/ui/query-builder/right-refiner-column/modal-options.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    return (
+      <div
+        ref={ref}
+        className="top-4 right-6 absolute z-50 text-14 font-normal"
+      >
+        <div className="top-8 flex flex-col justify-between px-0 py-0 bg-white rounded-md shadow-dark">
+          <div
+            onClick={(e: React.MouseEvent) => handleDeleteFilterBlock(e)}
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/modal-options.tsx]: y no onClick={handleDeleteFilterBlock}

File: src/pages/filter/ui/query-builder/right-refiner-column/modal-options.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            className="flex items-center justify-between py-2 px-2 rounded-br-none rounded-bl-none rounded-l-md rounded-r-md cursor-pointer hover:bg-blue-bright hover:text-white"
+            data-testId={DecisionTreeModalDataCy.joinByAnd}
+          >
+            <div className="mr-2">{t('filter.delete')}</div>
+
+            <Icon name="Delete" />
+          </div>
+
+          <div
+            onClick={(e: React.MouseEvent) => handleCopyFilterName(e)}
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/modal-options.tsx]: same

File: src/pages/filter/ui/query-builder/right-refiner-column/query-results.tsx [hunk 3]
```diff
@@ -30,16 +30,14 @@ export const QueryResults = observer((): ReactElement => {
 
     datasetStore.fetchDsStatAsync()
 
-    if (!datasetStore.isXL) {
-      datasetStore.fetchWsListAsync()
-    }
+    if (!datasetStore.isXL) datasetStore.fetchWsListAsync()
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/query-results.tsx]: keep curly braces pls
it's easier to add something if needed and make if blocks more readable

File: src/pages/filter/ui/query-builder/right-refiner-column/selected-filter-card.tsx [hunk 4]
```diff
@@ -30,34 +33,80 @@ export const SelectedFilterCard = observer(
...
+          className={cn(
+            'relative flex justify-between items-center border-b border-grey-light py-4 pl-3 pr-0.5 cursor-pointer',
+            isFilterActive && 'bg-blue-light',
+          )}
+          onClick={() => filterStore.setActiveFilterId(filterId)}
         >
-          <span className="leading-16px">{filterName}</span>
+          <div
+            className="flex"
+            onClick={(event: React.MouseEvent) =>
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/selected-filter-card.tsx]: same

File: src/pages/filter/ui/query-builder/right-refiner-column/selected-filter-card.tsx [hunk 5]
```diff
@@ -30,34 +33,80 @@ export const SelectedFilterCard = observer(
...
           <Icon
-            name="Arrow"
-            className={cn(
-              'text-blue-bright ml-auto transform transition-transform',
-              isOpen ? 'rotate-90' : '-rotate-90',
-            )}
+            name="Options"
+            className="cursor-pointer text-blue-bright flex justify-self-end"
+            stroke={false}
+            onClick={(event: React.MouseEvent) =>
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/selected-filter-card.tsx]: same

**Review by AlMaQntr — APPROVED** [review]

**Comment by ghost** [comment]
https://user-images.githubusercontent.com/87524295/160122636-9b8f4fd6-526b-4c3e-9a9b-428644059d55.mp4

**Comment by ghost** [comment]
Refiner crashes after certain actions (recorded on video)

**Review by AndreyUstyumenko — APPROVED** [review]

**Comment by AndreyUstyumenko** [comment]
@Lapk1n FYI

**Comment by Lapk1n** [comment]
Thanks! I will check that

**Comment by Lapk1n** [comment]
The last element must contain data about problem groups. Here what we get from backend

Screenshot 2022-03-28 at 18 00 54


No comments)

**Review by ghost — APPROVED** [review]

**Pull request merged**